sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -491,10 +493,14 @@ private[hive] class HiveMetastoreCatalog(val client: ClientInterface, hive: Hive
     result.copy(expectedOutputAttributes = Some(metastoreRelation.output))
   }
 
-  override def getTables(databaseName: Option[String]): Seq[(String, Boolean)] = {
+  override def getTables(databaseName: Option[String]): Seq[(String, Boolean)] = {hive
     val db = databaseName.getOrElse(client.currentDatabase)
+    client.listTables(db).map(tableName => getTable(db, tableName))
```

Comment:
@dilipbiswal  Yes, it works. The test failure does not happen in def client.listTables, but happens in action map(tableName => getTable(db, tableName)). In the procedure of map, a DROP TABLE action may occure, and then getTable you added throw exception.